buildSrc/src/main/java/io/micronaut/guides/core/GuideUtils.java

```diff
@@ -37,6 +40,42 @@ public static List<String> getTags(Guide guide) {
         return tagsList.stream().collect(Collectors.toList());
     }
 
+    public static List<String> getAppFeatures(App app, Language language) {
+        if (language == Language.JAVA) {
+            return mergeLists(app.features(), getAppInvisibleFeatures(app), app.javaFeatures());
+        }
+        if (language == Language.KOTLIN) {
+            return mergeLists(app.features(), getAppInvisibleFeatures(app), app.kotlinFeatures());
+        }
+        if (language == Language.GROOVY) {
+            return mergeLists(app.features(), getAppInvisibleFeatures(app), app.groovyFeatures());
+        }
+        return mergeLists(app.features(), getAppInvisibleFeatures(app));
+    }
+
+    public static List<String> getAppInvisibleFeatures(App app) {
+        if (app.validateLicense()) {
+            List<String> result = new ArrayList<>();
+            addAllSafe(result,app.invisibleFeatures());
```

Comment:
why are you calling `addAllSafe`, you know `result` is not null? You instantiated it in the previous line.